sky/packages/sky/lib/rendering/object.dart

```diff
@@ -376,10 +457,12 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
 
   static List<RenderObject> _nodesNeedingLayout = new List<RenderObject>();
   bool _needsLayout = true;
+  /// Whether this render object's layout information is dirty
   bool get needsLayout => _needsLayout;
   RenderObject _relayoutSubtreeRoot;
   bool _doingThisLayoutWithCallback = false;
   Constraints _constraints;
+  /// The layout constraints most recently supplied by the parent
   Constraints get constraints => _constraints;
   bool debugDoesMeetConstraints(); // override this in a subclass to verify that your state matches the constraints object
```

Comment:
document this one, since you have to override it if you implement a new RenderObject subclass